src/features/compare/compare.js

```diff
@@ -0,0 +1,28 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+    CompareProd: []
+}
+
+export const comparedProductSlice = createSlice({
+    name: 'CompareProd',
+    initialState,
+    reducers: {
+        addProductToCompare: (state, action) => {
+            const productIdToAdd = action.payload;
+            if (!state.CompareProd.includes(productIdToAdd)) {
+                state.CompareProd.length < 2 ?
+                    state.CompareProd = state.CompareProd.concat(productIdToAdd)
+                    :
+                    state.CompareProd = [productIdToAdd, state.CompareProd[0]]
```

Comment:
you mutate state again. this is not allowed in redux. from what I see you mutate your state almost in every reducer. check everything again and only after that I will review this PR again, it doesn't make sense to do it now, this is a very big mistake. fix it and I'll review again